src/core/constraint.jl

```diff
@@ -24,3 +24,105 @@ function constraint_tp_storage_loss(pm::PMs.GenericPowerModel, n::Int, i, bus, r
     JuMP.@NLconstraint(pm.model, sum(ps[c] for c in conductors) + (sd - sc) == standby_loss + sum( r[c]*(ps[c]^2 + qs[c]^2)/vm[c]^2 for c in conductors))
 end
 
+
+"""
+Also referred to as a constant power load.
+Fixes the load power sd.
+sd = [sd_1, sd_2, sd_3]
+What is actually fixed, depends on whether the load is connected in delta or wye.
+When connected in wye, the load power equals the per-phase power sn drawn at the
+bus to which the load is connected.
+sd_1 = v_a.conj(i_a) = sn_a
+When connected in delta, the load power gives the reference in the delta reference
+frame. This means
+sd_1 = v_ab.conj(i_ab) = (v_a-v_b).conj(i_ab)
+We can relate this to the per-phase power by
+sn_a = v_a.conj(i_a)
+    = v_a.conj(i_ab-i_ca)
+    = v_a.conj(conj(s_ab/v_ab) - conj(s_ca/v_ca))
+    = v_a.(s_ab/(v_a-v_b) - s_ca/(v_c-v_a))
+So for delta, sn is constrained indirectly.
+"""
+function constraint_tp_load_power_setpoint(pm::PMs.GenericPowerModel, load_id::Int; nw=pm.cnw)
+    load = PMs.ref(pm, nw, :load, load_id)
+    pd = load["pd"]
+    qd = load["qd"]
+    conn = load["conn"]
+    if conn=="wye"
+        for c in PMs.conductor_ids(pm)
+            constraint_load_power_setpoint_wye(pm, nw, c, load_id, pd[c], qd[c])
+        end
+    elseif conn=="delta"
+        constraint_tp_load_power_setpoint_delta(pm, nw, load_id, load["load_bus"], pd, qd)
+    else
+        Memento.error(LOGGER, "Unknown load connection type $conn.")
+    end
+end
+
+
+"""
+Also referred to as a constant current load.
+Sets the active and reactive load power sd to be proportional to
+the the voltage magnitude.
+pd = cp.|vm|
+qd = cq.|vm|
+sd = cp.|vm| + j.cq.|vm|
+The same remark applies on delta/wye as for the fixed setpoint.
+"""
+function constraint_tp_load_power_prop_vm(pm::PMs.GenericPowerModel, load_id::Int; nw=pm.cnw)
+    load = PMs.ref(pm, nw, :load, load_id)
+
+    vnom_kv = load["vnom_kv"]
+    vbase_kv_LL = PMs.ref(pm, nw, :bus, load["load_bus"])["base_kv"]
+    vbase_kv_LN = vbase_kv_LL/sqrt(3)
```

Comment:
To be consistent with PowerModels conventions, this load model should be in p.u. before getting to this point.

Reply:
If the vbase of the load is not saved, it has to be guaranteed that the bus to which the load is connected, and the load, have the same vbase. This is hard to do as long as the p.u. handling is spread out accross opendss.jl for the TPPM components, and delegated to PM for all PM components. For now, I propose to keep this here until p.u. is dealt with explicitly in one place for TPPMs.